### PR TITLE
Removed possible none from the justification of disabled recommendation

### PR DESCRIPTION
**Pull request by Fewwy**
The recommendation is turned off for 1 or more clusters
image

The recommendation is turned off for all clusters, the justification provided
image

The recommendation is turned off for all clusters, the justification isn't provided
image

### CODE REVIEW

**Comment by gkarat** [comment]
I think the text that you have in the "new" screenshot doesn't make sense. What about just having 
`"This recommendation has been disabled for all systems on 21 Feb 2022."`? Also pinging @doruskova

**Comment by doruskova** [comment]
@gkarat @Fewwy I was reviewing all empty states in RHEL a couple of months ago and there is a Jira ticket for some changes. It comes from this review https://docs.google.com/document/d/1rg_imN7ZrKDXG-cb6v3UOEzOsvKwdGzPcWpvO5_ijqg/edit?usp=sharing
It should be like this:
1a) This recommendation was disabled because [reason] and has no results.
1b) This recommendation was disabled and has no results.
2a) This recommendation was disabled for [2] clusters because [reason] and has no results. View clusters
2b) This recommendation was disabled for [2] clusters and has no results. View clusters

For all these use-cases we have a button "Enable this recommendation for all clusters"

Here's the Jira ticket. Advisor didn't change it yet https://issues.redhat.com/browse/ESSNTL-929

**Comment by doruskova** [comment]
@gkarat We agreed on this version with @Fewwy 
1a) This recommendation was disabled on 19 May 2022 and has no results.
1b) This recommendation was disabled on 19 May 2022 because [reason] and has no results.
2a)This recommendation was disabled for [2] clusters on 19 May 2022 and has no results. View clusters
2b) This recommendation was disabled for [2] clusters on 19 May 2022 because [reason] and has no results. View clusters

**Comment by Fewwy** [comment]
Currently, we couldn't show the reason/justification and date for the clusters in case they were turned off 1 by 1.
Consulted with Marie and updated the code and messages accordingly

File: compiled-lang/en.json [hunk 1]
```diff
@@ -81,11 +81,12 @@
   "ruleIsDisabledForAllClusters": "Recommendation is disabled for all clusters",
   "ruleIsDisabledForAllSystems": "Recommendation is disabled for all systems",
   "ruleIsDisabledForClusters": "Recommendation is disabled for some clusters",
-  "ruleIsDisabledForClustersBody": "Recommendation is disabled for {clusters, plural, one {# cluster} other {# clusters}}",
+  "ruleIsDisabledForClustersBody": "Recommendation is disabled for {clusters, plural, one {# cluster} other {# clusters}} and has no results",
```
Comment (gkarat) [inline compiled-lang/en.json:84]: A dot in the end of the sentence required. The same applies to other descriptors (I won't comment on them separately).

File: src/Components/Recommendation/Recommendation.js [hunk 2]
```diff
@@ -331,26 +331,44 @@ const Recommendation = ({ rule, ack, clusters, match }) => {
...
+                                          new Date(
+                                            ackData?.updated_at ||
+                                              ackData?.created_at
+                                          )
+                                        }
+                                        type="onlyDate"
+                                      />
+                                    </span>
+                                  ),
+                                  reason: ackData?.justification,
```
Comment (gkarat) [inline src/Components/Recommendation/Recommendation.js:351]: Nit-pick: `?` is not required (you already check the presence of `justification` in the upper ternary operator).

File: src/Components/Recommendation/Recommendation.js [hunk 3]
```diff
@@ -331,26 +331,44 @@ const Recommendation = ({ rule, ack, clusters, match }) => {
...
                                 }
-                                type="onlyDate"
-                              />
-                            </span>
-                          )}
+                              )
+                            : intl.formatMessage(
+                                messages.ruleIsDisabledWithoutJustificaiton,
+                                {
+                                  date: (
```
Comment (gkarat) [inline src/Components/Recommendation/Recommendation.js:357]: A nit-pick: you can keep the update date in a variable/constant somewhere above and just use it everywhere you need (not repeat yourself)

**Review by gkarat — APPROVED** [review]
Good job, please, add dots everywhere it is needed and we can merge it :-)

**Comment by gkarat** [comment]
@doruskova, yep, as Alex said, I am not sure of how we gonna show the common reason (justification) for a group of clusters for which a recommendation is disabled. Imagine we have 10 clusters hit by the recommendation. Users disable 3 of them with reason A, other three with reason B. Then, what the the card should show in that case?

File: src/Components/Recommendation/Recommendation.js (hunk 3 above)
Comment (Fewwy) [reply src/Components/Recommendation/Recommendation.js:357]: Done, created a const ruleDate for that

**Review by gkarat — APPROVED** [review]
OK, looks good to me, thank you.

**Comment by doruskova** [comment]
@gkarat @Fewwy Oohh, yes! Then we should say only "This recommendation was disabled for [2] clusters and has no results. View clusters"

And when you click on "View clusters" you should get a modal and the justification note is there. The same with a date. https://marvelapp.com/prototype/g70a1he/screen/81634580

**Pull request merged**